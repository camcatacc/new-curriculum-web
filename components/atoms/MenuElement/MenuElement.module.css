.linkButton:hover {
	color: #3ceb93 !important;
	font-weight: 600 !important;
	transition: color 0.2s ease-in;
}
