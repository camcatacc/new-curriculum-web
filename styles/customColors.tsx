export const black = "rgb(35,35,35)";
export const darkGrey = "rgb(112,112,112)";
export const lightGrey = "#E2E2E2";
export const primary = "#3CEB93";
export const secondary = "#000243";
export const white = "#FFFFFF";

export const brightPink = "#E25487";
export const darkPink = "#B2436B";
export const brightGreen = "#70C541";
export const darkGreen = "#60A13C";
export const brightBlue = "#5487E2";
export const darkBlue = "#406CB9";
