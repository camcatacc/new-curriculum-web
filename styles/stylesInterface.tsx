import CSS from "csstype";

export interface InlineStyles {
    [key: string]: CSS.Properties;
}
