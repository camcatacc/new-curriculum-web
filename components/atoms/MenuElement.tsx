import { Link, Typography } from "@material-ui/core";
import React from "react";
import { primary, white } from "../../styles/customColors";
import styles from "./MenuElement.module.css";

export const MenuElement = (props: {
    selected: boolean;
    onPressed: () => void;
    name: string;
}) => {
    const { name, selected, onPressed } = props;
    return (
        <Typography>
            <Link
                href=""
                color="textSecondary"
                style={{
                    color: selected ? primary : white,
                }}
                className={`cursor-pointer ${styles.linkButton} ${
                    selected ? "font-semibold" : "font-regular"
                }`}
                onClick={() => {}}
            >
                {name}
            </Link>
        </Typography>
    );
};
