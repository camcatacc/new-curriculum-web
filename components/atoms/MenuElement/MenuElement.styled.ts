// Modules
import styled from "styled-components";

// Elements
import { Link as MaterialLink } from "@material-ui/core";

// Styles
import { primary } from "styles/customColors";

export const Link = styled(MaterialLink)<{ $selected: boolean; $color?: string }>`
	color: ${({ $selected, $color }) => ($selected ? primary : $color ?? "white")};
	:hover {
		color: #3ceb93 !important;
		font-weight: 600 !important;
		transition: color 0.2s ease-in;
	}
`;
