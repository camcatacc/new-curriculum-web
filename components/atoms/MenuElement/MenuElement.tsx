// Modules
import React from "react";

// Elements
import { Link, Typography } from "@material-ui/core";

// Styles
import { primary, white } from "styles/customColors";
import styles from "./MenuElement.module.css";

// Definitions
export interface MenuElementProps {
	selected: boolean;
	onClicked: () => void;
	title: string;
}

// Element
const MenuElement = ({ selected, onClicked, title }: MenuElementProps) => (
	<Typography>
		<Link
			color="textSecondary"
			style={{
				color: selected ? primary : white
			}}
			className={`cursor-pointer ${styles.linkButton} ${selected ? "font-semibold" : "font-regular"} uppercase`}
			onClick={onClicked}
		>
			{title}
		</Link>
	</Typography>
);

export default MenuElement;
