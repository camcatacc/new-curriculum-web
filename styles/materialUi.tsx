import { createTheme } from "@material-ui/core/styles";
import { primary, secondary } from "styles/customColors";

export const getMaterialTheme = (paletteType: "light" | "dark") =>
	createTheme({
		palette: {
			type: paletteType,
			primary: { main: primary, dark: secondary, light: primary },
			secondary: { main: secondary, dark: primary, light: secondary }
		},
		typography: {
			fontFamily: ['"Arial"'].join(","),
			h2: { fontFamily: ['"Segoe UI"'].join(","), fontWeight: 700, fontSize: "1.6rem" },
			h3: { fontWeight: 600, fontSize: "1.1rem" },
			body1: { fontWeight: 400, fontSize: "1.2rem" },
			body2: { fontWeight: 200, fontSize: "1rem" }
		}
	});
