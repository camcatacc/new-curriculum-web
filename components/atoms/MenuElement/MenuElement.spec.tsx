// Modules
import React from "react";
import { fireEvent, render } from "@testing-library/react";

// Elements
import MenuElement from "components/atoms/MenuElement/MenuElement";

// Styling
import { primary } from "styles/customColors";

// Definitions
import type { MenuElementProps } from "components/atoms/MenuElement/MenuElement";

// Default Props
const defaultProps: MenuElementProps = {
	onClicked: jest.fn(),
	selected: false,
	title: "Home"
};

// Rendering
const renderComponent = (props = {}) => {
	const combinedProps = { ...defaultProps, ...props };
	return render(<MenuElement {...combinedProps} />);
};

// Tests
describe("MenuElement", () => {
	it("Renders as expected", () => {
		const { container } = renderComponent();
		expect(container).toMatchSnapshot();
	});

	it("If selected is not true, the color is white", () => {
		const { getByRole } = renderComponent({ selected: false });
		const link = getByRole("link");
		expect(link).toHaveStyle("color: rgb(255,255,255)");
	});

	it("If selected is true, the color is primary", () => {
		const { getByRole } = renderComponent({ selected: true });
		const link = getByRole("link");
		expect(link).toHaveStyle(`color: ${primary}`);
	});

	it("If clicked, call the onClick function", () => {
		const { getByRole } = renderComponent();
		const link = getByRole("link");
		fireEvent.click(link);
		expect(defaultProps.onClicked).toHaveBeenCalled();
	});
});
