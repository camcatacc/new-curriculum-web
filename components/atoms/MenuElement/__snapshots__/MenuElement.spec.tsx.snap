// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuElement Renders as expected 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body2"
  >
    <a
      class="MuiTypography-root MuiLink-root MuiLink-underlineHover sc-bdvvtL nvKmn cursor-pointer font-regular uppercase MuiTypography-colorTextSecondary"
    >
      Home
    </a>
  </p>
</div>
`;
